bot/cogs/codeblock/instructions.py

```diff
@@ -0,0 +1,177 @@
+"""This module generates and formats instructional messages about fixing Markdown code blocks."""
+
+import logging
+from typing import Optional
+
+from . import parsing
+
+log = logging.getLogger(__name__)
+
+_EXAMPLE_PY = "{lang}\nprint('Hello, world!')"  # Make sure to escape any Markdown symbols here.
+_EXAMPLE_CODE_BLOCKS = (
+    "\\`\\`\\`{content}\n\\`\\`\\`\n\n"
+    "**This will result in the following:**\n"
+    "```{content}```"
+)
+
+
+def _get_example(language: str) -> str:
+    """Return an example of a correct code block using `language` for syntax highlighting."""
+    # Determine the example code to put in the code block based on the language specifier.
+    if language.lower() in parsing.PY_LANG_CODES:
+        log.trace(f"Code block has a Python language specifier `{language}`.")
+        content = _EXAMPLE_PY.format(lang=language)
+    elif language:
+        log.trace(f"Code block has a foreign language specifier `{language}`.")
+        # It's not feasible to determine what would be a valid example for other languages.
+        content = f"{language}\n..."
+    else:
+        log.trace("Code block has no language specifier.")
+        content = "\nHello, world!"
+
+    return _EXAMPLE_CODE_BLOCKS.format(content=content)
+
+
+def _get_bad_ticks_message(code_block: parsing.CodeBlock) -> Optional[str]:
+    """Return instructions on using the correct ticks for `code_block`."""
+    log.trace("Creating instructions for incorrect code block ticks.")
+
+    valid_ticks = f"\\{parsing.BACKTICK}" * 3
```

Comment:
Should I completely remove the constant then? I feel it helps with readability.

Reply:
I think it helps in cases like here https://github.com/python-discord/bot/blob/201895180ffbe88c01e4dbc40dd9cd6c043e2be7/bot/cogs/codeblock/parsing.py#L86-L88 but imo a literal is better here and in `_TICKS` when it's already a string (or around other strings) and with the var name hinting on what it is if the small char is hard to read for some.

I see where you're coming from though so some other opinions would be beneficial

Reply:
FWIW, I find the current version with a constant easier to read:

```py
valid_ticks = f"\\{parsing.BACKTICK}" * 3
```

Compared to using a literal:
```py
valid_ticks = f"\\`" * 3
```

It's a bit difficult to make out the backtick unless you're specifically looking for it. Depending on the font and zoom setting used, this could be even harder to see.

Reply:
Using the constant improves readability for me as well. I think the main thing that looks "weird", is the combination of an escaped slash with an f-string placeholder. It looks like there's much more going on than just creating a string with a backslash and a grave accent.

Maybe turning into a raw f-string would help, although I have no strong opinion on that:

```py
# with constant
valid_ticks = fr"\{parsing.BACKTICK}" * 3
# without constant
valid_ticks = fr"\`" * 3
```